### ISSUE TITLE
Patreon?

### ISSUE THREAD

**Issue by viktor-ku** [19majkel94/type-graphql#151]
Hi,

I'd like to support this project with Patreon. Even though I am ready to donate no more than $5 a month maybe you could get more people to support it as well?

What do you think about it?

Thanks for your project :heart:

**Comment by 19majkel94** [comment]
Thanks!
Actually, I haven't thought about it. It's a great idea but I'm not sure if it's the time do do that. So if they are more people ready to donate, please 👍 the first post to let me know that you want to support the project 😉 

And I think that Open Collective is a better place to make a financial contributions to open source projects:
https://opencollective.com/

**Comment by viktor-ku** [comment]
@19majkel94 yeah, the platform doesn't matter. I just want this project running but I don't have enough time to contribute with code so I figured...

**Comment by anodynos** [comment]
:+1: :+1: :+1 I'm already donating a tenner on the projects I use, type-graphql deserves a few more :-)

**Comment by tylergets** [comment]
Also would be interested in supporting from our company, we utilize the project in production and would also love to keep it alive :)

**Comment by anodynos** [comment]
Great news to know @tylergets your company using it in production!

**Comment by Veetaha** [comment]
Really nice codebase you have :+1:. Unfortunately, I am just a student and possess no money sources rather than my scholarship and parents' donations  :unamused:, but all your `typestack` libraries are very awesome to use in my projects and I wish I could contribute to your library :heart::heart:. I wish you gain more popularity than any other backend framework :muscle::muscle:

**Comment by 19majkel94** [comment]
> all your typestack libraries are very awesome

Thanks but they are not mine - @pleerock is the author, I'm only one of the contributors 😉  

>  I am just a student and possess no money sources

You can help TypeGraphQL by spreading the word, sharing on twitter, writing blog posts, tutorials, sample projects on Github and many more 💪 

I am starting with TypeGraphQL page on opencollective.com this month (I hope so), so all interested bakers will soon be able to support the project with donations 🎉

**Comment by viktor-ku** [comment]
I've used it in one of my projects that are currently in production. Unfortunately I'am switching over to the `rust` for my next server app, but will support this project nonetheless :heart:

**Comment by anodynos** [comment]
I 've just started donating $24/year :-)

I'm happy to be the first one apparently https://opencollective.com/angelospikoulas ! 

@19majkel94 please add https://opencollective.com/19majkel94 to your home page, the github readme.md & everywhere!

And lets all contribute all we can, especially those using it commercially, to make this project even greater!

**Comment by 19majkel94** [comment]
@viktor-ku @tylergets @Veetaha
You can now support TypeGraphQL development efforts 🎉 
All the donation can be made using the Open Collective platform:
https://opencollective.com/typegraphql

@anodynos 
I've modified the first post, pinned this issue, added backers section in readme and donate button in top of the readme, as well as the support section in the website landing page 💪 

You can now spread the word and like/retweet 🐦 
https://twitter.com/typegraphql/status/1109830853906042880

**Comment by goldcaddy77** [comment]
Happy to join as a member!  Thanks again @19majkel94 for all of the hard work on this project!!

**Comment by 19majkel94** [comment]
@goldcaddy77 Thank you very much! 😍

**Comment by MichalLytek** [comment]
Quote by @andreimc:
https://github.com/MichalLytek/type-graphql/issues/553
> I just wanted to do a shout out yo you @MichalLytek  and the rest of the contributors, you guys have helped us a lot and we have started contributing a monthly $50 dollars, we know it's not a lot but we are also a start up! Once we get more traction we will become a Gold Sponsor.
>
> 👍 and keep up the good work! We are loving this project and it has saved us numerous hours!
>
> A big shout out from the Mr Yum team  and lots of ❤️ from Melbourne, Australia!

**Comment by MichalLytek** [comment]
Quote by @avin-kavish moved from #574 ❤️ 

![image](https://user-images.githubusercontent.com/10618781/80921105-6d6ffd00-8d74-11ea-85a0-40d53ee66c06.png)

**Comment by StarpTech** [comment]
We also plan to sponsor this project but it's not transparent what we can expect in the near future. According to project (TypeGraphQL vNext) and commit history it looks like there is no progress for months (Also can't see a Next branch) and that throwing money on it will have a marginal or effectively zero impact for us. Could you elaborate on your commitment to this project in 2021?

**Comment by MichalLytek** [comment]
> (Also can't see a Next branch)

I've moved that into a private repo because Kamil Myśliwiec "stolen" my code and embedded it as `nestjs-graphql`.
I don't want to share this ideas anymore before the `v2.0` will be ready.

> looks like there is no progress for months 

I can see plenty of not released yet features:
https://github.com/MichalLytek/type-graphql/blob/master/CHANGELOG.md#unreleased

`v1.2` needs some polishing and some other pending fixes and PRs before releasing. I see a lot of people still uses `v0.17.6` so I try to publish bigger updates less frequently, so they do not need to bump deps every week.

I'm in a process of moving into a new house so temporarily there might be less commit activity because I only have time to answer issues, discussion and stackoverflow. Most of my engagement is now focused on [TypeGraphQL Prisma integration](https://prisma.typegraphql.com/).

**Comment by StarpTech** [comment]
Thank you for the fast response.

> I've moved that into a private repo because Kamil Myśliwiec "stolen" my code and embedded it as nestjs-graphql.
I don't want to share those vNext ideas anymore before the v2.0 will be ready.

This is no opensource 😕 How is it possible to steal from an MIT licensed project? Please rethink that decision.

> I can see plenty of not released yet features:
https://github.com/MichalLytek/type-graphql/blob/master/CHANGELOG.md#unreleased

Any release ETA?

> I'm in a process of moving into a new house so temporarily there might be less commit activity because I only have time to answer issues, discussion and stackoverflow. Most of my engagement is now focused on TypeGraphQL Prisma integration.

Congrats! 🏡 We would like to see improvements in the core. We are also willing to contribute in form of code.

- https://github.com/MichalLytek/type-graphql/issues/12
- https://github.com/MichalLytek/type-graphql/issues/183

**Comment by MichalLytek** [comment]
> How is it possible to steal from an MIT licensed project? Please rethink that decision.

Imagine you spend a lot of time working on a new OSS project. Let's say a new NoSQL database. You work very hard for months, working in silent, having a 100 GitHub stars. Then goes Microsoft, "steal" (copy the code or fork the repo), name it "MicrosoftDB", launch in Azure, launch marketing campaigns, tutorial, influencers, etc.

Now everyone is using your project under some other company/project name. People forget about what you've done (or never heard of the origin of "MicrosoftDB", you don't have sponsors anymore. Everyone is using the Microsoft product, you only see your name in some file headers.

Welcome in a OSS MIT world 😉 
Read about the Redis vs. AWS case to see how others monetize someone's OSS projects:
https://techcrunch.com/2019/02/21/redis-labs-changes-its-open-source-license-again/

> take their products and sell them as rebranded services without contributing back to the community

**Comment by StarpTech** [comment]
I created a ton of OOS. Some are more popular than others. I can understand you but the conclusion of such use cases was not to get away from opensource they changed their license to avoid such tricky shoots. Your project and great achievements are still visible and recognized.

**Comment by StarpTech** [comment]
> Welcome in a OSS MIT world 😉
Read about the Redis vs. AWS case to see how others monetize someone's OSS projects:

That's a pitfall. If your goal is to make money, MIT or anything in general without a business plan is 💩 Someday somebody will eat you 😄 The motivation to drive this project should be business-driven or personal.

**Comment by MichalLytek** [comment]
> If your goal is to make money

The goal is not to make money. The goal is to be able to work full-time on OSS. And that requires variety of ways to raise funds for doing this. We all have mortgages, family, so we can't do it pro-bono, like we were still 23 and studying.

That's the reality, without the funding, it becomes a maintenance only project, as the first wave of excitement, gaining popularity and GitHub stars is putting out...

So having sponsors is really helping me in reducing my daily-job hours, so I have time to spend for working on TypeGraphQL 😉 

> Your project and great achievements are still visible and recognized.

Go ask NestJS users if they know who created the code-first, decorator based GraphQL implementation they use.
Nobody knows, nobody remembers. It hurt me very bad, so vNext is now developed in secret and be released as MIT when it's done - not as some partial ideas that can be implemented by "the competitor" in the meantime, when I don't have enough time to work on them.

**Comment by StarpTech** [comment]
As I said before, I understand! But if you develop the software behind the eyes of the opensource community, it will leave a resentful residue.
